esp/services/ws_workunits/ws_workunitsService.cpp

```diff
@@ -5014,3 +5014,442 @@ bool CWsWorkunitsEx::onWUGetArchiveFile(IEspContext &context, IEspWUGetArchiveFi
     }
     return true;
 }
+
+//Copy from eclcmd_common.cpp
+bool CWsWorkunitsEx::isValidPriorityValue(const char *value)
+{
+    if (isEmptyString(value))
+        return false;
+    if (strieq("SLA", value) || strieq("LOW", value) || strieq("HIGH", value) || strieq("NONE", value))
+        return true;
+    return false;
+}
+
+//Copy from eclcmd_common.cpp
+bool CWsWorkunitsEx::isValidMemoryValue(const char *value)
+{
+    if (isEmptyString(value) || !isdigit(*value))
+        return false;
+    while (isdigit(*++value));
+
+    if (!*value)
+        return true;
+
+    switch (toupper(*value++))
+    {
+        case 'E':
+        case 'P':
+        case 'T':
+        case 'G':
+        case 'M':
+        case 'K':
+            if (!*value || strieq("B", value))
+                return true;
+            break;
+        case 'B':
+            if (!*value)
+                return true;
+            break;
+    }
+    return false;
+}
+
+const char *CWsWorkunitsEx::readQueryFileCopyErrors(IArrayOf<IConstLogicalFileError> &errors, StringBuffer &errorMsg)
```

Comment:
I'm not too fond of the name "readQueryFileCopyErrors" your really sort of gathering them into a StringBuffer.. so maybe "gatherQueryFileCopyErrors"?